feat: Option to Add Organization Logo in PDF Generated Report #52

dockabernathy01 opened the pull request:
feat: Option to Add Organization Logo in PDF Generated Report #52

-  Feature.

Issue Number: #52

I have added a option to add organization logo in PDF.

dacoaster reviewed (changes requested):
This lacks any way for them to choose or upload the logo, so it isn’t finished yet.

Please add the option to enable and upload a logo on the `ReportsTab.vue` in the settings menu.

It should have a slider toggle to enable which then allows them to access a file selector input.

This will require a new key titled “reportLogo” in the `config.json` and default configuration. The settings menu should update this config with the path to the file they choose.

You can then remove the checkbox from the settings panel and remove logo from the session.

dockabernathy01 commented:
@dacoaster . I have fixed the ui as you mentioned.
Please check and let me know.
And from now on, I will start with the new issue.

dockabernathy01 commented:
@dacoaster . I have fixed the ui as you mentioned.
Please check and let me know.
And from now on, I will start with the new issue.

dacoaster commented on src/assets/new_logo.png:
Remove this file used for testing.

dacoaster commented on src/assets/temp_logo.png:
Remove this file used for testing

dacoaster commented on src/components/settings/ReportsTab.vue:
```diff
@@ -28,12 +28,64 @@
+        <v-divider class="" width="200px"></v-divider>
+        <v-img
+          class="bg-white"
+          :src="require('../../assets/new_logo.png')"
```
This shouldn’t be hard coded.

the src should be a `file://` link if `config.reportLogo` is set.

Like [here](https://github.com/yatt-ai/yattie/blob/3a1f1e12cd96ce3be7d3482caa06fb62391c652f/src/views/PrintView.vue#L113)

dacoaster commented on src/components/settings/ReportsTab.vue:
```diff
@@ -66,6 +118,23 @@ export default {
+        if (status === STATUSES.ERROR) {
+          this.$root.$emit("set-snackbar", message);
+        } else {
+          // const currentPath = this.$router.history.current.path;
```
This function doesn’t seem to do anything when successful?

we don’t need a new ipc function to open images, just use a the `src` attribute with a `file://` path like we do [everywhere else](https://github.com/yatt-ai/yattie/blob/3a1f1e12cd96ce3be7d3482caa06fb62391c652f/src/views/PrintView.vue#L113).

dockabernathy01 replied:
I have added the action when successful.

dacoaster commented on src/modules/FileSystemUtility.js:
```diff
@@ -399,3 +398,53 @@ module.exports.openExternalLink = async (url = "") => {
+  });
+};
+
+module.exports.openImage = async () => {
```
This function is unnecessary and should be removed.

dockabernathy01 replied:
I need to use openImage function to get the path of image.

dacoaster commented on src/views/PrintView.vue:
```diff
@@ -1,7 +1,7 @@
-    <div class="header">
-      <img :src="require('../assets/logo.png')" />
+    <div class="header" v-if="reportLogo">
+      <img :src="require('../assets/new_logo.png')" />
```
This shouldn’t be hardcoded. Please see comments above

dacoaster commented:
@dockabernathy01 - see requested changes

dockabernathy01 commented:
@dacoaster , As you can see in the code, I have saved the logo to new_logo image.
And I used this new_logo image in PrintView.
But If you want to change, I will change this.

dockabernathy01 commented:
@dacoaster , I have changed the code as you mentioned.
Please check and let me know.
Thanks.

dacoaster commented on src/components/ExportPanel.vue:
```diff
@@ -131,8 +132,9 @@ export default {
         timer: this.$store.state.session.timer,
         started: this.$store.state.session.started,
         ended: this.$store.state.session.ended,
+        reportLogo: this.config.reportLogo,
+        logoPath: this.config.logoPath,
```
If we are going to have multiple fields, we should nest them:

```
reports: {
  logo: {
    enabled: true,
    path: "/home/user/logo.png"
  }
}
```

dockabernathy01 replied:
Ok, got it.

dacoaster commented on src/components/settings/ReportsTab.vue:
```diff
@@ -66,6 +118,22 @@ export default {
+        if (status === STATUSES.ERROR) {
+          this.$root.$emit("set-snackbar", message);
+        } else {
+          this.config.logoPath = path;
+          this.$emit("submit-config", this.config);
+        }
+      } else {
+        // todo Add web version handler
+      }
```
@dockabernathy01 - If we just use a standard [`input`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file) for this, then we can get the file path directly from the that element _and_ it will work for both web and electron.  Then we can remove all of the `$electronService` specific code.

dockabernathy01 replied:
I'm sorry. it will work on web version too.
I couldn't remove that.
It works.

dacoaster replied:
I would like this done with a standard HTML file input.

dockabernathy01 replied:
Currently, I used electron dialog.
You mean, you want me to use v-file input?

dockabernathy01 replied:
Like this.
const {  dialog,  BrowserWindow } = require("electron");
dialog.showOpenDialog()

dacoaster replied:
Yes, please use a `v-file-input` in the view above and have the image card src display the value of that input.

dockabernathy01 replied:
Ok, I will change like that.
But I think, it doesn't look good in ui side.

dockabernathy01 replied:
And in addition, why don't you use Slack or MS Teams for discussing about the work?(Of course, we will do contract on upwork.)
I think it's more flexible to discuss about issues.

dockabernathy01 commented:
@dacoaster , I have fixed the code as you mentioned.
I have removed openImage and added v-file-input.
And I have finished feat: quicker download of report PDF #129 .
After merging this PR, I will make PR with #129 .
Thanks.

dacoaster commented on src/components/ExportPanel.vue:
```diff
@@ -132,8 +132,8 @@ export default {
-        reportLogo: this.config.reportLogo,
-        logoPath: this.config.logoPath,
+        reportLogo: logo && logo.enabled,
+        logoPath: logo && logo.path && logo.path.path,
```
@dockabernathy01 - what is `logo.path.path`? I don’t see the nested `path` variables anywhere else

dockabernathy01 replied:
@dacoaster , I have stored the path like this : 
    "path": {
      "path": "E:\\Mine\\Electron\\Yattie-Session\\logo.png",
      "name": "logo.png",
      "size": 1610
    }

dockabernathy01 replied:
![logo](https://github.com/yatt-ai/yattie/assets/146516559/8292905e-834e-4179-b3b4-b1821976df1c)

dacoaster commented on src/components/settings/ReportsTab.vue:
```diff
@@ -60,11 +112,32 @@ export default {
   },
+  created() {
+    if (!this.config.logo) {
+      this.config.logo = {
```
@dockabernathy01 Rather than adding this here, can you add the logo data to the default config like mentioned [here](https://github.com/yatt-ai/yattie/pull/146#discussion_r1534244078)

dockabernathy01 replied:
I will update this.

dacoaster commented on src/components/settings/ReportsTab.vue:
```diff
@@ -60,11 +112,32 @@ export default {
+    if (!this.config.logo) {
+      this.config.logo = {
+        enabled: false,
+        path: {},
```
@dockabernathy01 Shouldn’t `path` be an empty string?

dockabernathy01 replied:
@dacoaster , I stored the logo path like this : 
 , I have stored the path like this : 
    "path": {
      "path": "E:\\Mine\\Electron\\Yattie-Session\\logo.png",
      "name": "logo.png",
      "size": 1610
    }
So, I think it will be empty object.

dacoaster replied:
`name` and `size` don't belong under `path`.

It should be:
```
logo: {
    enabled: BOOLEAN,
    path: STRING,
    name: STRING,
    size: INT
}
````

dockabernathy01 replied:
@dacoaster , Ok, I will update the code like that.

dockabernathy01 commented:
@dacoaster , Can I check in your computer using Anydesk?
